fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -217,27 +223,24 @@ struct MutationRef {
 	                    Arena& arena,
 	                    BlobCipherMetrics::UsageType usageType,
 	                    StringRef* buf = nullptr) const {
-		const BlobCipherEncryptHeader* header = encryptionHeader();
-		auto textCipherItr = cipherKeys.find(header->cipherTextDetails);
-		auto headerCipherItr = cipherKeys.find(header->cipherHeaderDetails);
-		ASSERT(textCipherItr != cipherKeys.end() && textCipherItr->second.isValid());
-		ASSERT(headerCipherItr != cipherKeys.end() && headerCipherItr->second.isValid());
-		TextAndHeaderCipherKeys textAndHeaderKeys;
-		textAndHeaderKeys.cipherHeaderKey = headerCipherItr->second;
-		textAndHeaderKeys.cipherTextKey = textCipherItr->second;
+		TextAndHeaderCipherKeys textAndHeaderKeys = getCipherKeys(cipherKeys);
 		return decrypt(textAndHeaderKeys, arena, usageType, buf);
 	}
 
 	TextAndHeaderCipherKeys getCipherKeys(
-	    const std::unordered_map<BlobCipherDetails, Reference<BlobCipherKey>>& cipherKeys) {
+	    const std::unordered_map<BlobCipherDetails, Reference<BlobCipherKey>>& cipherKeys) const {
 		const BlobCipherEncryptHeader* header = encryptionHeader();
-		auto textCipherItr = cipherKeys.find(header->cipherTextDetails);
-		auto headerCipherItr = cipherKeys.find(header->cipherHeaderDetails);
-		ASSERT(textCipherItr != cipherKeys.end() && textCipherItr->second.isValid());
-		ASSERT(headerCipherItr != cipherKeys.end() && headerCipherItr->second.isValid());
+		auto getCipherKey = [&](const BlobCipherDetails& details) -> Reference<BlobCipherKey> {
+			if (!details.isValid()) {
+				return {};
```

Comment:
Is this acceptable? Should we add a log?

Reply:
yes, for the header cipher (see previous comment). we shall keep the assert for text cipher key though. will do in separate PR.